controllers/configurationpolicy_controller.go

```diff
@@ -479,23 +479,30 @@ func (r *ConfigurationPolicyReconciler) shouldEvaluatePolicy(
 	var interval time.Duration
 	var getIntervalErr error
 
-	if policy.Status.ComplianceState == policyv1.Compliant && policy.Spec != nil {
-		interval, getIntervalErr = policy.Spec.EvaluationInterval.GetCompliantInterval()
-	} else if policy.Status.ComplianceState == policyv1.NonCompliant && policy.Spec != nil {
-		interval, getIntervalErr = policy.Spec.EvaluationInterval.GetNonCompliantInterval()
-	} else {
+	switch policy.Status.ComplianceState {
+	case policyv1.Compliant:
+		if policy.Spec != nil {
+			interval, getIntervalErr = policy.Spec.EvaluationInterval.GetCompliantInterval()
+		}
+	case policyv1.NonCompliant:
+		if policy.Spec != nil {
+			interval, getIntervalErr = policy.Spec.EvaluationInterval.GetNonCompliantInterval()
+		}
+	case policyv1.UnknownCompliancy, policyv1.Terminating:
```

Comment:
I think this is going to give a different result with the `policy.Spec` check nested.

```suggestion
	if policy.Spec != nil {
		switch policy.Status.ComplianceState {
		case policyv1.Compliant:
			interval, getIntervalErr = policy.Spec.EvaluationInterval.GetCompliantInterval()
		case policyv1.NonCompliant:
			interval, getIntervalErr = policy.Spec.EvaluationInterval.GetNonCompliantInterval()
		}
	} else {
```

Reply:
question why `case policyv1.UnknownCompliancy, policyv1.Terminating:
			log.V(1).Info("The policy has an unknown compliance. Will evaluate it now.")

			return true, 0
		}` is unknown compliance?

Reply:
Oh, good point. But actually, I tried out a `nil` spec (which seemed like an odd case), and the controller logs that it's recovering from a panic. Looks like the original goal when it was put in place was to be able to determine whether it was set, but really we should require it instead:
- https://github.com/open-cluster-management-io/config-policy-controller/pull/84

I'll open another PR to address it.

Reply:
PR: https://github.com/open-cluster-management-io/config-policy-controller/pull/349

Reply:
Oops, sorry--my PR is going to force a rebase here. Hopefully it's not too painful! 😅